### ISSUE TITLE
Update sql.y to support index-related storage parameter queries

### ISSUE THREAD

**Issue by nickvigilante** [cockroachdb/cockroach#84906]
It would be nice for our customers to have diagrams within our docs outlining how to set and reset storage parameters surrounding indexes. The following commands should be supported:

- ALTER INDEX ... SET STORAGE PARAMETER
- ALTER INDEX ... RESET STORAGE PARAMETER

Jira issue: CRDB-17942

**Comment by ajwerner** [comment]
Is this asking for diagrams or is this asking for support for SQL syntax? I don't see us supporting the syntax here, so I don't think this issue is just asking for diagram support. For now moving this to experience to support that syntax on indexes. If it turns out we do support the syntax, I think the issue still belongs there.

**Comment by nickvigilante** [comment]
Sounds good. Yes, we ultimately want to create diagrams for setting and resetting storage parameters on indexes, but that won't come until the syntax is added to the `sql.y` file. It should be possible given the documentation we have on spatial indexes.

**Comment by ajwerner** [comment]
Are you saying that this issue is a placeholder for syntax diagrams for syntax cockroach currently does not support? We support storage parameters on spatial indexes, but I don't think we support changing them.

**Comment by nickvigilante** [comment]
No, this issue is for the `sql.y` support and changing storage parameters on indexes. We can create the diagrams in a different ticket (and it should be either my responsibility or docs team's responsibility).

**Comment by ajwerner** [comment]
I feel like we're talking past each-other. Support in sql.y doesn't mean anything except the parser accepts it. That's useless if the database does not support it. The database does not support it. So, is this issue asking for the database to support it?

**Comment by nickvigilante** [comment]
Ok, I think the disconnect was I assumed the DB supported it. If it's not supported, then this ticket should serve to update the database to support it, followed by the parser.

**Comment by ajwerner** [comment]
👍 we do not currently support changing storage parameters for existing indexes. We only support setting them when creating new indexes.
